akka-http/src/main/scala/akka/http/scaladsl/server/directives/SecurityDirectives.scala

```diff
@@ -254,7 +256,14 @@ trait SecurityDirectives {
    *
    * @group security
    */
-  def challengeFor(realm: String) = HttpChallenge(scheme = "Basic", realm = realm, params = Map.empty)
+  def basicChallengeFor(realm: String) = HttpChallenge(scheme = "Basic", realm = realm, params = Map.empty)
```

Comment:
> private?

Yeah, it seems like they should be used within `SecurityDirectives`.

> Just add the scheme as a param and keep a single method?

You mean something like this and remove the methods I've defined?
```scala
def challengeFor(scheme: String, realm: String) = HttpChallenge(scheme = scheme, realm = realm, params = Map.empty)
```

I, personally, prefer the way I did as there might be reuse of the methods, such as `def basicChallengeFor(...)`. Currently we don't reuse either of them, and your idea would be suitable :)